### ISSUE TITLE
Azure platform automerge selects a merge strategy for a different target branch

### ISSUE THREAD

**Issue by chrisvanderpennen** [renovatebot/renovate#21639]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.55.1

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the bug

When [populating `config.defaultMergeMethod`](https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/azure/index.ts#L206-L212), the Azure platform module fetches all policy configurations from the project.  Because it passes a null `branchRef` parameter to `getMergeMethod`, `isRelevantScope` returns every policy matching the current repository regardless of target branch.  getMergeMethod then [picks the first one](https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/azure/azure-helper.ts#L150), which in our case is not the main branch of the repository.  I'm not sure why this is or how to cause Azure DevOps to behave this way for a reproduction, but I believe I have identified a resolution.

When creating the PR, the Azure platform module then [blindly applies the default merge method](https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/azure/index.ts#L457-L474), which in our case is not valid for the default branch.

I believe this could be simply fixed by borrowing the [merge method lookup code](https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/azure/index.ts#L686-L693) from `mergePr` in the same module.  As the PR is created before autocomplete is applied, the targetRefName will be populated.

### Relevant debug logs

```
DEBUG: ProjectName/RepoName default branch = develop (repository=ProjectName/RepoName)
TRACE: getProjectAndRepo(ProjectName/RepoName) (repository=ProjectName/RepoName)
TRACE: getMergeMethod(01234567-89ab-cdef-0123-456789abcdef, ProjectName, null) determining mergeMethod from matched policy:
       {
           "allowRebase": true,
           "allowRebaseMerge": true,
           "scope": [
               {
                   "refName": "refs/heads/release",
                   "matchKind": "Prefix",
                   "repositoryId": "01234567-89ab-cdef-0123-456789abcdef"
               }
           ]
       } (repository=ProjectName/RepoName)
```
In our case. policy attached to develop does not permit rebases, and instead is as follows:
```
{
    "allowSquash": true,
    "allowRebaseMerge": true,
    "scope": [
        {
            "refName": "refs/heads/develop",
            "matchKind": "Exact",
            "repositoryId": "01234567-89ab-cdef-0123-456789abcdef"
        }
    ]
}
```
### Have you created a minimal reproduction repository?

~~I have explained in the description why a minimal reproduction is impossible~~

I have been able to reproduce this at https://dev.azure.com/chrisvanderpennen/public/_git/renovate-21639-repro

**Comment by chrisvanderpennen** [comment]
I've been able to reproduce this at https://dev.azure.com/chrisvanderpennen/public/_git/renovate-21639-repro - it appears the policies are returned by the Azure DevOps API in creation order.

Renovate has created [this pull request](https://dev.azure.com/chrisvanderpennen/public/_git/renovate-21639-repro/pullrequest/10) with an autocompletion strategy of a basic merge, which is not permitted by the policy on the main branch and the only thing preventing it from merging.

Debug level build logs are available [here](https://dev.azure.com/chrisvanderpennen/public/_build/results?buildId=10&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=35f53584-79fc-539f-463c-6471d0663a3a), I can capture and attach them here if you'd prefer.  I can also set up another run with trace level logging if you'd like.

**Comment by rarkins** [comment]
Checking I understand right.

You have set platformAutomerge=true

Renovate has created a PR with automerge (as expected)

Azure prints these details:

<img width="841" alt="image" src="https://user-images.githubusercontent.com/6311784/234475985-e2d3ee72-0196-4484-8be2-eb03006aa64a.png">

Here is Renovate's logic for this:

https://github.com/renovatebot/renovate/blob/7f4c8ebeb919091b4773c177fa30f96873c6ac95/lib/modules/platform/azure/index.ts#L457-L474

Here is where Renovate gets the default merge method for the repo: https://github.com/renovatebot/renovate/blob/7f4c8ebeb919091b4773c177fa30f96873c6ac95/lib/modules/platform/azure/index.ts#L206-L212

You're saying that instead of using a default merge method for the repo, Renovate should check each individual branch/PR after creation before setting platform automerge, using same logic as https://github.com/renovatebot/renovate/blob/7f4c8ebeb919091b4773c177fa30f96873c6ac95/lib/modules/platform/azure/index.ts#L686-L693 ?

**Comment by chrisvanderpennen** [comment]
Yes, that's what I am proposing.

The merge method set as the "default" for the repo is the first one that matches the provided `repoId` and `targetRefName` in the document returned by the policy API, or if no `branchRef` is provided then it falls back to the first policy in the document that sets a merge type on _any_ branch in that repo:
https://github.com/renovatebot/renovate/blob/7f4c8ebeb919091b4773c177fa30f96873c6ac95/lib/modules/platform/azure/azure-helper.ts#L143-L149

https://github.com/renovatebot/renovate/blob/7f4c8ebeb919091b4773c177fa30f96873c6ac95/lib/modules/platform/azure/azure-helper.ts#L123-L141

Here, the Azure DevOps API is returning the merge policy I created matching `release/*` before the policy assigned to `main`.  They appear to be sent by the API in the order in which they were created.

Would it help if I attached the API response here?
